### PR TITLE
Restructure EnergyPlusData; Move AirflowNetworkBalanceManager data to data struct

### PR DESCRIPTION
**Pull request by mitchute**
This PR reorganizes the data classes and moves the ANBM data to a data class in the ANBM.hh file with the ANBM state data instance living in the ANBM.cc file (for now). The pattern established for ANBM is an incremental step that should be replicated for all remaining state data. Once all actual state data has been cleaned up (by creating ```enum```'s, ```constexpr```'s, removing unused ```extern```'s, etc) and moved to a data class, the data class instances can be moved to EnergyPlusData. This should allow the code to be cleaned up to some degree and allow for reduced build times.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp]: @mitchute This is creating an `#include` cycle, which is not failing in this case but I'd generally try to avoid. It'd be better to pull the enums into this file than to include the manager header.

File: src/EnergyPlus/AirflowNetworkBalanceManager.hh
Comment (jasondegraw) [inline src/EnergyPlus/AirflowNetworkBalanceManager.hh:218]: @mitchute When I split the implementation up into the current subdirectory form, my intent was to circle back and handle some of this stuff at a later date. I'd propose that instead of naming this struct `AirflowNetworkBalanceManagerData` that we name it `AirflowNetworkBalanceManager` instead and get rid of the `AirflowNetworkBalanceManager` namespace. That may leave some of the objects in a bit of limbo right now, but we can clean that up later.

Comment (mitchute) [reply src/EnergyPlus/AirflowNetworkBalanceManager.hh:218]: @jasondegraw If all of these functions were member methods of an object class, then I think that could work. But this would leave all of this hanging out on the ```EnergyPlus``` namespace. I'm not sure if this is going to be problematic. I'll defer to others on this one.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp
Comment (mitchute) [reply src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp]: OK, I can do that.

**Comment by mitchute** [comment]
@Myoldmopar @jasondegraw feel free to take a look at this when you get a chance. Jason suggested dropping the ANBM namespace, which is still an open question.

**Comment by jasondegraw** [comment]
@mitchute In my mind it depends on your intent. If this is to be a "complete" pattern to follow elsewhere, then it makes sense to me to make those free functions into member functions, and the namespace serves little to no purpose. If this is just meant to be a step in the right direction, then the namespace does serve a purpose containing those free functions and it should be kept. I'm fine with it either way, I'll be back in AFN next month and can do some refactoring then.

**Comment by mitchute** [comment]
> @mitchute In my mind it depends on your intent. If this is to be a "complete" pattern to follow elsewhere, then it makes sense to me to make those free functions into member functions, and the namespace serves little to no purpose. If this is just meant to be a step in the right direction, then the namespace does serve a purpose containing those free functions and it should be kept. I'm fine with it either way, I'll be back in AFN next month and can do some refactoring then.

@jasondegraw Understood. This is just an incremental step so I'm gonna leave it as-is for now. Feel free to push it in that direction when you do your refactoring.

**Comment by mitchute** [comment]
@Myoldmopar I believe this is ready to go.

File: src/EnergyPlus/AirflowNetworkBalanceManager.hh
Comment (Myoldmopar) [reply src/EnergyPlus/AirflowNetworkBalanceManager.hh:218]: I think we should make one dedicated pass on the airflow network side of things, and do it next, right away.  I'd like for this chunky PR to get dropped in as-is if possible.

**Comment by Myoldmopar** [comment]
@mitchute @jasondegraw thanks for the effort and review on this.  CI is almost entirely happy.  I deleted the Mac results just now and will let it run again to make sure the failure was an anomaly.  I'll pull develop in locally and build and run tests just to be sure everything is happy and give one more once-over on the code changes.  Assuming all is well, I think this can merge this morning.

File: src/EnergyPlus/AirflowNetworkBalanceManager.hh
Comment (jasondegraw) [reply src/EnergyPlus/AirflowNetworkBalanceManager.hh:218]: @Myoldmopar OK, I'll start up a housekeeping branch with some miscellaneous fixes, leaving the manager alone. Once this is merged, I'll pull back in develop and work on the manager a bit. It shouldn't take too long to get into shape to go in.

File: src/EnergyPlus/CMakeLists.txt
Comment (Myoldmopar) [inline src/EnergyPlus/CMakeLists.txt:86]: Indentation gone awry

File: src/EnergyPlus/CommandLineInterface.cc
Comment (Myoldmopar) [inline src/EnergyPlus/CommandLineInterface.cc:388]: Is this change actually just in this branch?  I thought the GIO branch already went in.  I think maybe Github is just confused.

File: src/EnergyPlus/DElightManagerF.hh
Comment (Myoldmopar) [inline src/EnergyPlus/DElightManagerF.hh:55]: I'm not sure which branch brought this change in, but it would be great to know if it was intentionally a quote style include rather than a bracket style.

**Review by Myoldmopar — APPROVED** [review]
I'm really excited about where this is going.  I understand that while some of those deep nested function calls still need to be access lots of global data, this could cause a temporary impact on build times, but as those dependencies get cleaned out, it will get back to just depending on explicitly what it needs to depend on.

It's unfortunate that Github is showing this Files Changed tab with all the GIO changes mixed in.  I can generally see which changes where added for the restructure, and I'm OK with them.  One minor indentation tweak in the CMakeLists file won't hold this up.

File: src/EnergyPlus/DElightManagerF.hh
Comment (jasondegraw) [reply src/EnergyPlus/DElightManagerF.hh:55]: Well, since we should be using quotes rather than brackets, I'd take this as an improvement... where is that bikeshed emoji when you need it...

Comment (Myoldmopar) [reply src/EnergyPlus/DElightManagerF.hh:55]: :bike: :classical_building:

**Comment by mitchute** [comment]
> It's unfortunate that Github is showing this Files Changed tab with all the GIO changes mixed in.

I'm working on another globals refactor that I based on this branch. Do you want me to start over from a branch based on develop?

**Comment by Myoldmopar** [comment]
No, just don't open a PR until this PR is closed.  That seems to be the point when Github creates the cached files changed baseline.

**Comment by Myoldmopar** [comment]
Built locally with develop pulled in, running tests now.  Assuming it is happy, I think this could be fine to go in.  It will cause some conflicts probably, but hopefully not too many.

**Comment by Myoldmopar** [comment]
Mac returned happy on the re-run, so this is good to go.  This won't start causing diffs in results from other branches, but it has the potential for conflicts.  My plan is to get as many other branches merged in this afternoon as I can, then resolve all the conflicts at once in this branch, then merge this in tonight.

If anyone is reviewing a branch and feel like it's basically ready, I'd urge you to be a little more aggressive and try to get them in this afternoon to avoid having to resolve conflicts in extra branches.

**Comment by Myoldmopar** [comment]
And we're good to go.  Merging.  Thanks @mitchute and @jasondegraw.  This is a small step in a really awesome direction.

**Pull request merged**
